src/chunk.c

```diff
@@ -833,6 +852,14 @@ set_complete_chunk(ChunkScanCtx *scanctx, Chunk *chunk)
 	return false;
 }
 
+static bool
+chunk_scan_context_add_chunk(ChunkScanCtx *scanctx, Chunk *chunk)
```

Comment:
What I meant was that the scan context already contains the chunks you need, so instead of returning a new "container" of the chunks "extracted" from the context, just return the context. So, you would do like follows:

For each hypertable:
1. Scan for chunks -> return scan context
1. Add the size of the scan context (# hash table entries) to a cumulative `num_chunks` count.
1. Add scan context to a list

Once done:
1. Allocate a `Chunk *` array of size `num_chunks`
1. Iterate each scan context  and add each chunk to the `Chunk *` array
1. Sort the `Chunk *` array

This would avoid the need for the `Chunks` boilerplate code